controller_manager/src/controller_manager.cpp

```diff
@@ -334,43 +303,48 @@ void ControllerManager::subscribe_to_robot_description_topic()
   RCLCPP_INFO(
     get_logger(), "Subscribing to '%s' topic for robot description.",
     robot_description_subscription_->get_topic_name());
+
+  // Setup diagnostics
+  diagnostics_updater_.setHardwareID("ros2_control");
+  diagnostics_updater_.add(
+    "Controllers Activity", this, &ControllerManager::controller_activity_diagnostic_callback);
+
+  robot_description_notification_timer_ = create_wall_timer(
+    std::chrono::seconds(1),
+    [&]()
+    {
+      RCLCPP_WARN(
+        get_logger(), "Waiting for data on '~/robot_description' topic to finish initialization");
+    });
```

Comment:
```suggestion
// This is needed to start the CM services when the URDF is already loaded
  if (resource_manager_->is_urdf_already_loaded())
  {
    init_services();
  }
  else
  {
      robot_description_notification_timer_ = create_wall_timer(
        std::chrono::seconds(1),
        [&]()
        {
          RCLCPP_WARN(
            get_logger(), "Waiting for data on '~/robot_description' topic to finish initialization");
        });
    }
```

Reply:
Actually, you are right! but I think that the solution is rather to initialize the timer before subscriber. Then we have less code and it should work in both cases.